### PR TITLE
Aut 8658

### PR DESCRIPTION
**Pull request by lieblichaaron**
## ℹ️ Change description:
Nothing here

<details><summary><sub>templates<sub></summary>
<p>
  
<sub>[templates: [bug](?expand=1&template=bug.md), [feature](?expand=1&template=feature.md)]</sub>

</p>
</details>

### CODE REVIEW

File: examples/client/Locomotion/src/Components/BsPages/index.tsx [hunk 1]
```diff
@@ -309,6 +313,33 @@ export const CancelRide = (props: any) => {
   );
 };
 
+export const ConfirmFutureRide = (props: any) => {
+  const { newFutureRide } = useContext(FutureRidesContext);
+  const theme = useContext(ThemeContext);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/BsPages/index.tsx]: You aren't using theme

File: examples/client/Locomotion/src/Components/BsPages/index.tsx [hunk 2]
```diff
@@ -3,15 +3,20 @@ import {
...
 import DatePicker from 'react-native-date-picker';
+import TextRowWithIcon from '../../Components/TextRowWithIcon';
+import { FutureRidesContext } from '../../context/futureRides';
+import { STOP_POINT_TYPES } from '../../lib/commonTypes';
 import SvgIcon from '../SvgIcon';
 import { RidePageContext } from '../../context/newRideContext';
 import i18n from '../../I18n';
-import { ERROR_COLOR, FONT_SIZES, FONT_WEIGHTS } from '../../context/theme';
+import {
+  Context as ThemeContext, ERROR_COLOR, FONT_SIZES, FONT_WEIGHTS,
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/BsPages/index.tsx]: why you change the ThemeContext?

File: examples/client/Locomotion/src/Components/BsPages/index.tsx [hunk 3]
```diff
@@ -309,6 +313,33 @@ export const CancelRide = (props: any) => {
...
+  const dropOffText = i18n.t('bottomSheetContent.confirmFutureRide.dropOffText', { address: dropOff?.description });
+
+  return (
+    <BsPage
+      TitleText={i18n.t('bottomSheetContent.confirmFutureRide.titleText')}
+      ButtonText={i18n.t('bottomSheetContent.confirmFutureRide.buttonText')}
+      fullWidthButtons
+      {...props}
+    >
+      <TextRowWithIcon text={dateText} icon={timeIcon} />
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/BsPages/index.tsx]: extract to function and create the variables there

File: examples/client/Locomotion/src/Components/BsPages/index.tsx [hunk 4]
```diff
@@ -309,6 +313,33 @@ export const CancelRide = (props: any) => {
...
+      <TextRowWithIcon text={pickupText} />
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/BsPages/index.tsx]: same

File: examples/client/Locomotion/src/Components/BsPages/index.tsx [hunk 5]
```diff
@@ -309,6 +313,33 @@ export const CancelRide = (props: any) => {
...
+      <TextRowWithIcon text={dropOffText} />
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/BsPages/index.tsx]: same

File: examples/client/Locomotion/src/Components/Marker/index.js [hunk 6]
```diff
@@ -19,10 +22,11 @@ const StopPointMarker = ({
   stopPoint,
   key,
   isNext,
-  chosenService,
+  etaText,
+  isFutureRide,
 }) => {
   const { lat, lng } = stopPoint;
-
+  const theme = useContext(ThemeContext);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/Marker/index.js:29]: use styled-component ThemeContext

File: examples/client/Locomotion/src/Components/TextRowWithIcon/index.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
+import React, { useContext } from 'react';
+import { Text, View } from 'react-native';
+import styled from 'styled-components';
+import { FONT_SIZES, Context as ThemeContext } from '../../context/theme';
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/TextRowWithIcon/index.tsx]: use ThemeContext from styled-component

File: examples/client/Locomotion/src/context/bottomSheetContext/index.js [hunk 8]
```diff
@@ -19,12 +19,14 @@ export const SNAP_POINT_STATES = {
   [BS_PAGES.CUSTOM_TIP]: ['99.999999%', '100%'],
   [BS_PAGES.LOCATION_REQUEST]: ['30%'],
   [BS_PAGES.CANCEL_RIDE]: ['30%'],
+  [BS_PAGES.CONFIRM_FUTURE_RIDE]: ['45%'],
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/bottomSheetContext/index.js]: use static height, we don't want difference between different devices

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/RideButtons/FutureBookingButton/index.tsx [hunk 9]
```diff
@@ -18,8 +18,8 @@ const FutureBookingButton = () => {
   const { ride } = useContext(RidePageContext);
   const nowText = i18n.t('bottomSheetContent.ride.now');
   const { primaryColor } = useContext(ThemeContext);
-  const isFutureRide = ride?.afterTime;
-  const chosenTime = isFutureRide && moment(ride.afterTime).format('DD.MM.YY, HH:mm');
+  const isFutureRide = ride?.scheduledTo;
+  const chosenTime = isFutureRide && moment(ride.scheduledTo).format('DD.MM.YY, HH:mm');
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/RideButtons/FutureBookingButton/index.tsx:23]: const chosenTime = ride?.scheduledTo && moment(ride.scheduledTo).format('DD.MM.YY, HH:mm');

Comment (lieblichaaron) [reply examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/RideButtons/FutureBookingButton/index.tsx:23]: why? i use it in multiple places

File: examples/client/Locomotion/src/pages/ActiveRide/newMap.js [hunk 10]
```diff
@@ -133,6 +139,15 @@ export default React.forwardRef(({
         });
       }
     }
+    if (currentBsPage === BS_PAGES.CONFIRM_FUTURE_RIDE) {
+      ref.current.fitToCoordinates(newFutureRide.stopPoints.map(sp => ({
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/newMap.js]: extract the fitToCoodinations to function and send the array, or create the array and call the function at the end of the main function

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/RideButtons/FutureBookingButton/index.tsx (hunk 9 above)
Comment (avivbarniv) [reply examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/RideButtons/FutureBookingButton/index.tsx:23]: my bad, my view is just the changes

**Pull request merged**